### PR TITLE
Add practice exercise `minesweeper`

### PR DESCRIPTION
**Pull request by jiegillet**
New one, pretty standard :)

### CODE REVIEW

File: exercises/practice/minesweeper/.meta/tests.toml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# As user-added comments (using the # character) will be removed when this file
+# is regenerated, comments can be added via a `comment` key.
+
+[0c5ec4bd-dea7-4138-8651-1203e1cb9f44]
+description = "no rows"
+
+[650ac4c0-ad6b-4b41-acde-e4ea5852c3b8]
+description = "no columns"
+include = false
+comment = "empty board already covered by 0c5ec4bd-dea7-4138-8651-1203e1cb9f44"
```
Comment (jiegillet) [inline exercises/practice/minesweeper/.meta/tests.toml:18]: This test was transforming `"\n"` into `"\n"`, but in my implementation, `"\n"` is considered an empty board that outputs an empty string. I felt that this was an unnecessary edge case to consider.

File: exercises/practice/minesweeper/test/minesweeper_test.gleam [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+pub fn space_surrounded_by_mines_test() {
+  minesweeper.annotate(
+    "***
+*_*
+***",
+  )
+  |> should.equal(
+    "***
+*8*
+***",
```
Comment (jiegillet) [inline exercises/practice/minesweeper/test/minesweeper_test.gleam:62]: This doesn't look super nice, but introducing a leading newline in the output would be weird.

**Review by lpil — APPROVED** [review]
What a fun exercise! Thank you
